packages/muelu/test/unit_tests/Smoothers/BlockedDirectSolver.cpp

```diff
@@ -321,51 +304,87 @@ namespace MueLuTests {
-
-      TEUCHOS_TEST_COMPARE(residualNorm2[0], <, 5e-15, out, success);
-      TEUCHOS_TEST_COMPARE(finalNorms[0] - Teuchos::ScalarTraits<Scalar>::magnitude(Teuchos::ScalarTraits<Scalar>::one()), <, 5e-15, out, success);
-
-      if (comm->getSize() == 1) {
-        TEST_EQUALITY(residualNorm1[0] == residualNorm2[0], true);
-      } else {
-        out << "Pass/Fail is only checked in serial." << std::endl;
+      Array<magnitude_type> norms(1);
+      X->norm2(norms);
+      X->scale(1 / norms[0]);
+
+      for(const auto& solver_type : solver_types)
+      {
+        //////////////////////////////////////////////////////////////////////
+        // Smoothers
+        RCP<BlockedDirectSolver> smootherPrototype = rcp(new BlockedDirectSolver(solver_type));
+        RCP<SmootherFactory> smootherFact = rcp(new SmootherFactory(smootherPrototype));
+
+        // main factory manager
+        FactoryManager M;
+        M.SetFactory("Smoother", smootherFact);
+
+        MueLu::SetFactoryManager SFM(Teuchos::rcpFromRef(level), Teuchos::rcpFromRef(M));
+
+        // request BGS smoother (and all dependencies) on level
+        level.Request("Smoother", smootherFact.get());
+        level.Request("PreSmoother", smootherFact.get());
+        level.Request("PostSmoother", smootherFact.get());
+
+        smootherFact->Build(level);
+
+        RCP<SmootherBase> solver = level.Get<RCP<SmootherBase> >("PreSmoother", smootherFact.get());
+
+        // Compute RHS corresponding to X
+        A->apply(*X, *RHS, Teuchos::NO_TRANS, (SC) 1.0, (SC) 0.0);
+
+        // Reset X to 0
+        X->putScalar((SC) 0.0);
+
+        RHS->norm2(norms);
+        out << "||RHS|| = " << std::setiosflags(std::ios::fixed) << std::setprecision(10) << norms[0] << std::endl;
+
+        out << "solve with zero initial guess" << std::endl;
+        Teuchos::Array<magnitude_type> initialNorms(1);
+        X->norm2(initialNorms);
+        out << "  ||X_initial|| = " << std::setiosflags(std::ios::fixed) << std::setprecision(10) << initialNorms[0]
+            << std::endl;
+
+        solver->Apply(*X, *RHS, true);  //zero initial guess
+
+        Teuchos::Array<magnitude_type> finalNorms(1);
+        X->norm2(finalNorms);
+        Teuchos::Array<magnitude_type> residualNorm1 = Utilities::ResidualNorm(*A, *X, *RHS);
+        out << "  ||Residual_final|| = " << std::setiosflags(std::ios::fixed) << std::setprecision(20)
+            << residualNorm1[0] << std::endl;
+        out << "  ||X_final|| = " << std::setiosflags(std::ios::fixed) << std::setprecision(10) << finalNorms[0]
+            << std::endl;
+
+        TEUCHOS_TEST_COMPARE(residualNorm1[0], <, 5e-15, out, success);
+        TEUCHOS_TEST_COMPARE(
+                finalNorms[0] - Teuchos::ScalarTraits<Scalar>::magnitude(Teuchos::ScalarTraits<Scalar>::one()), <,
+                5e-15, out, success);
+
+        out << "solve with random initial guess" << std::endl;
+        X->randomize();
```

Comment:
Would there be additional value in prescribing the random seed to make this more reproducible?

Reply:
The seed is set above, this should be fine.